tests/parallel_test_core.py

```diff
@@ -27,10 +27,21 @@ def run(self):
 # run tests for one mode
 def run_mode(args):
   mode = args[0]
-  print '<< running %s >>' % mode
+  if len(args) > 1:
+    # If args has multiple elements, then only the selected tests from that suite are run e.g. args=['asm1', 'test_hello_world', 'test_i64'] runs only tests asm1.test_hello_world and asm1.test_i64.
+    for i in range(1, len(args)):
+      if args[i].startswith('test_'):
+        args[i] = mode + '.' + args[i]
+    args = args[1:]
```

Comment:
Hmm, not sure if it matters, since the code still needs to loop by index `i`, because it wants to change the contents. So the loop would become `for i in range(len(args))`. Although I suppose that might be a bit more idiomatic at least, so let me do that change and push.